com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs

```diff
@@ -35,68 +35,96 @@ public enum BehaviorType
     /// The Factory to generate policies.
     /// </summary>
     [AddComponentMenu("ML Agents/Behavior Parameters", (int)MenuGroup.Default)]
-    internal class BehaviorParameters : MonoBehaviour
+    public class BehaviorParameters : MonoBehaviour
     {
-        [HideInInspector]
-        [SerializeField]
+        [HideInInspector, SerializeField]
         BrainParameters m_BrainParameters = new BrainParameters();
-        [HideInInspector]
-        [SerializeField]
-        NNModel m_Model;
-        [HideInInspector]
-        [SerializeField]
-        InferenceDevice m_InferenceDevice;
-        [HideInInspector]
-        [SerializeField]
 
-        // Disable warning /com.unity.ml-agents/Runtime/Policy/BehaviorParameters.cs(...):
-        //   warning CS0649: Field 'BehaviorParameters.m_BehaviorType' is never assigned to,
-        //   and will always have its default value
-        // This field is set in the custom editor.
-        #pragma warning disable 0649
-        internal BehaviorType m_BehaviorType;
-        #pragma warning restore 0649
-        [HideInInspector]
-        [SerializeField]
-        string m_BehaviorName = "My Behavior";
+        /// <summary>
+        /// The associated <see cref="BrainParameters"/> for this behavior.
+        /// </summary>
+        public BrainParameters brainParameters
```

Comment:
Should we maybe rename the class BrainParameters? We do not even have Brains anymore...

Reply:
Open to suggestions - what would you call them?

Reply:
Discussed on slack, going ahead with this for now.